File: math612/tbnotes/matlab/script/TB_Lecture_25-26.m

A = magic(4)
lam = eig(A)

[X,D] = eig(A)

A = eye(4) + diag([1 1 1],1)   % Jordan block
[X,D] = eig(A);
rank_X = rank(X)
residual = norm(A*X-X*D)

A = eye(2) + 1e-10*[0 0;0 1]
norm( eig(A) - diag(A) )

norm( roots(poly(A)) - diag(A) )

A = magic(5)
[Q,T] = schur(A);
norm( Q'*Q-eye(5) )
T

A = rot90(A);
lam = eig(A)

[Q,T] = schur(A);
T

eig(T(2:3,2:3))
eig(T(4:5,4:5))

A

x = A(2:5,1);  v = norm(x)*eye(4,1)-x; F = eye(4)-2*(v*v')/(v'*v);

[A(1,:);F*A(2:5,:)]

F*A*F'









H = hess(A)

eig(H)

A = A+A';
hess(A)

File: math612/tbnotes/julia/script/TB_Lecture_25-26.jl

A = [
    16     2     3    13
     5    11    10     8
     9     7     6    12
     4    14    15     1

];
(λ,X) = eig(A);
@show λ;
@show X;

A = eye(4) + diagm([1;1;1],1)   # Jordan block
(λ,X) = eig(A);
@show rank_X = rank(X)

D = diagm(λ);
residual = norm(A*X-X*D)

A = [
    17    24     1     8    15
    23     5     7    14    16
     4     6    13    20    22
    10    12    19    21     3
    11    18    25     2     9
];

(T,Q) = schur(A);
T

A = A[:,end:-1:1]';
λ = eigvals(A)

(T,Q) = schur(A);
T

eigvals(T[2:3,2:3])

H = hessfact(A)[:H]

eigvals(H)

A = A+A';
hessfact(A)[:H]


